pkg/pool-weighted/contracts/smart/ManagedPool.sol

```diff
@@ -349,6 +352,100 @@ contract ManagedPool is BaseWeightedPool, ReentrancyGuard {
         emit SwapEnabledSet(swapEnabled);
     }
 
+    /**
+     * @dev Remove a token from the pool. This is a permissioned function that performs the following operations:
+     * - Verify there are enough tokens, and there is no ongoing weight change
+     * - Calculate the BPT value of the full token balance, and returns it for possible use by the caller
+     * - Exit the pool, returning the full balance of the token to the recipient
+     * - Deregister the token in the Vault (since it now has zero balance)
+     * - Reduce the total weight by the weight of the token removed, and update the number of tokens and `_tokenState`
+     *   (all other weights then scale accordingly)
+     */
+    function removeToken(uint256 tokenIndex, address recipient) external authenticate whenNotPaused returns (uint256) {
+        return _removeToken(tokenIndex, recipient);
+    }
+
+    function _removeToken(uint256 tokenIndex, address recipient) internal returns (uint256) {
+        // Calculate the BPT value of the full token balance (other values returned to avoid duplication)
+        //(uint256 bptAmountIn, uint256 minAmountOut, uint256 normalizedWeight) = getTotalBptForToken(tokenIndex);
```

Comment:
```suggestion
```

Reply:
That was leftover from something; removed